--- origsrc/xscreensaver-5.06/configure.in	2008-07-12 12:08:03.000000000 -0500
+++ src/xscreensaver-5.06/configure.in	2008-07-18 12:21:26.734375000 -0500
@@ -723,6 +723,7 @@
   # Look for the directory under a standard set of common directories.
   # Check X11 before X11Rn because it's often a symlink to the current release.
   for ac_dir in                                 \
+    /etc/X11/app-defaults                       \
     /usr/X11/lib/app-defaults                   \
     /usr/X11R6/lib/app-defaults                 \
     /usr/X11R6/lib/X11/app-defaults             \
--- origsrc/xscreensaver-5.06/driver/subprocs.c	2008-07-02 23:04:41.000000000 -0500
+++ src/xscreensaver-5.06/driver/subprocs.c	2008-07-18 12:21:26.765625000 -0500
@@ -718,7 +718,7 @@
 {
 #ifdef SIGCHLD
 
-# ifdef HAVE_SIGACTION	/* Thanks to Tom Kelly <tom@ancilla.toronto.on.ca> */
+# if defined(HAVE_SIGACTION) && !defined(__CYGWIN__)	/* Thanks to Tom Kelly <tom@ancilla.toronto.on.ca> */
 
   static Bool sigchld_initialized_p = 0;
   if (!sigchld_initialized_p)
--- origsrc/xscreensaver-5.06/driver/windows.c	2008-07-05 20:40:42.000000000 -0500
+++ src/xscreensaver-5.06/driver/windows.c	2008-07-18 12:21:26.765625000 -0500
@@ -747,7 +747,7 @@
 static void
 catch_signal (saver_info *si, int sig, RETSIGTYPE (*handler) (int))
 {
-# ifdef HAVE_SIGACTION
+# if defined(HAVE_SIGACTION) && !defined(__CYGWIN__)
 
   struct sigaction a;
   a.sa_handler = handler;
